src/elf/symver.rs

```diff
@@ -59,876 +59,866 @@
-        #[inline]
-        pub fn get_at(&self, idx: usize) -> Option<Versym> {
-            let entsize = core::mem::size_of::<ElfVersym>();
-            let offset = idx.checked_mul(entsize)?;
+        self.bytes.len() / entsize
+    }
 
-            self.bytes
-                .pread_with::<ElfVersym>(offset, self.ctx.le)
-                .ok()
-                .map(|ElfVersym { vs_val }| Versym { vs_val })
-        }
+    /// Get [`Versym`] entry at index.
+    #[inline]
+    pub fn get_at(&self, idx: usize) -> Option<Versym> {
+        let entsize = core::mem::size_of::<ElfVersym>();
+        let offset = idx.checked_mul(entsize)?;
+
+        self.bytes
+            .pread_with::<ElfVersym>(offset, self.ctx.le)
+            .ok()
+            .map(|ElfVersym { vs_val }| Versym { vs_val })
     }
+}
 
-    impl<'a> IntoIterator for &'_ VersymSection<'a> {
-        type Item = <VersymIter<'a> as Iterator>::Item;
-        type IntoIter = VersymIter<'a>;
+impl<'a> IntoIterator for &'_ VersymSection<'a> {
+    type Item = <VersymIter<'a> as Iterator>::Item;
+    type IntoIter = VersymIter<'a>;
 
-        fn into_iter(self) -> Self::IntoIter {
-            VersymIter {
-                bytes: self.bytes,
-                offset: 0,
-                ctx: self.ctx,
-            }
+    fn into_iter(self) -> Self::IntoIter {
+        VersymIter {
+            bytes: self.bytes,
+            offset: 0,
+            ctx: self.ctx,
         }
     }
+}
 
-    /// Iterator over the [`Versym`] entries from the [`SHT_GNU_VERSYM`] section.
-    pub struct VersymIter<'a> {
-        bytes: &'a [u8],
-        offset: usize,
-        ctx: container::Ctx,
-    }
+/// Iterator over the [`Versym`] entries from the [`SHT_GNU_VERSYM`] section.
+pub struct VersymIter<'a> {
+    bytes: &'a [u8],
+    offset: usize,
+    ctx: container::Ctx,
+}
 
-    impl<'a> Iterator for VersymIter<'a> {
-        type Item = Versym;
+impl<'a> Iterator for VersymIter<'a> {
+    type Item = Versym;
 
-        fn next(&mut self) -> Option<Self::Item> {
-            if self.offset >= self.bytes.len() {
-                None
-            } else {
-                self.bytes
-                    .gread_with::<ElfVersym>(&mut self.offset, self.ctx.le)
-                    .ok()
-                    .map(|ElfVersym { vs_val }| Versym { vs_val })
-                    .or_else(|| {
-                        // self.bytes are not a multiple of ElfVersym.
-                        // Adjust offset to continue yielding Nones.
-                        self.offset = self.bytes.len();
-                        None
-                    })
-            }
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset >= self.bytes.len() {
+            None
+        } else {
+            self.bytes
+                .gread_with::<ElfVersym>(&mut self.offset, self.ctx.le)
+                .ok()
+                .map(|ElfVersym { vs_val }| Versym { vs_val })
```

Comment:
nit: It would be nicer imho to implement `From<ElfVersym> for Versym` then you could do `.map(Versym::from)` I believe.

Reply:
Good idea, will do that